agent/websockets/connection.go

```diff
@@ -142,12 +146,10 @@ func NewConnection(ctx context.Context, targetURL string, header http.Header, er
 		}
 	}()
 	go func() {
-		wg.Wait()
-		if err := serverConn.WriteMessage(
-			websocket.CloseMessage,
-			websocket.FormatCloseMessage(websocket.CloseNormalClosure, "")); err != nil {
-			errCallback(fmt.Errorf("failure issuing a close message on a server websocket connection: %v", err))
-		}
```

Comment:
Can you explain why it's safe to remove this? Does the client expect a close message?

Reply:
The reason was that the server should handle closing the underlying TCP connection the same as it handles a close message. The close message becomes redundant . And in cases when this code is executed after a writing or reading error, it would have failed anyway. I have tested it with JupyterLab and it seems to be the case.

But you bring up a good point. For other servers, their implementation could be doing a different clean up when the close message arrives vs when the TCP connection is lost (which they shouldn't IMO). For those clients, I added code to send the closing message. But moved to Connection.Close so that it wouldn't cause failures if the closure was initiated of a lost TCP connection.

Reply:
PTAL at newly added code in next commit